lib/puppet/type/firewall.rb

```diff
@@ -234,6 +234,24 @@
     newvalues(%r{^\d+[[:graph:][:space:]]+$})
   end
 
+  newparam(:onduplicaterulebehaviour) do
+    desc <<-PUPPETCODE
+      In certain situations it is possible for an unmanaged rule to exist
+      on the target system that has the same comment as the rule
+      specified in the manifest.
+
+      This setting determines what happens when such a duplicate is found.
+
+      It offers three options:
+
+        * ignore - The duplicate rule is ignored and the run will continue.
+        * warn - The duplicate rule is logged as a warning and the run will continue.
+        * error - The duplicate rule is logged as an error and the run will fail.
```

Comment:
Suggest clarifying that the _resource_ will be updated (if necessary) vs "the run"